genie-web/src/test/java/com/netflix/genie/web/apis/rest/v3/controllers/JobRestControllerTest.java

```diff
@@ -628,6 +631,70 @@ void wontForwardJobOutputRequestIfOnCorrectHost() throws GenieException, GenieCh
             );
     }
 
+    @Test
+    void wontThrow404ForJobOutputRequestIfAgentNotFound() throws GenieException, GenieCheckedException {
+        this.jobsProperties.getForwarding().setEnabled(true);
+        final String jobId = UUID.randomUUID().toString();
+        final HttpServletRequest request = Mockito.mock(HttpServletRequest.class);
+        final HttpServletResponse response = Mockito.mock(HttpServletResponse.class);
+
+        Mockito
+            .when(request.getRequestURL())
+            .thenReturn(new StringBuffer("https://" + this.hostname + "/api/v3/jobs/1234/output"));
+
+        Mockito.when(this.persistenceService.getJobArchiveStatus(jobId)).thenReturn(ArchiveStatus.PENDING);
+        Mockito.when(this.persistenceService.isV4(jobId)).thenReturn(true);
+        Mockito.when(this.agentRoutingService.getHostnameForAgentConnection(jobId)).thenReturn(Optional.empty());
+
+        Assertions.assertThatThrownBy(
+            () -> this.controller.getJobOutput(jobId, null, request, response)
+        ).isInstanceOf(GenieServerException.class);
+
+        Mockito.verify(this.agentRoutingService, Mockito.times(1)).getHostnameForAgentConnection(Mockito.eq(jobId));
+    }
+
+    @Test
+    void wontForwardJobOutputRequestIfForwardingRequestWasForwarded() throws GenieException, GenieCheckedException {
```

Comment:
could use `@DisplayName` for these JUnit 5 tests to shorten the actual method names if you want FYI